PlutoHelperAgent/AppDelegate.m

```diff
@@ -222,56 +171,71 @@ -    (void)getUrl:(NSAppleEventDescriptor *)event
-        NSString *projectid = [parts objectAtIndex:2];
-        
-    
-        [ProjectLockerAndKeychainFunctions check_logged_in:^(enum ReturnValues connectionStatus) {
+        NSArray *pathsWhiteListArray = [[NSUserDefaults standardUserDefaults] arrayForKey:@"paths_list"];
+        BOOL pathGood = false;
+        for (id item in pathsWhiteListArray) {
+            if ([item[@"path"] isNotEqualTo:@""]) {
+                if ([projectPath hasPrefix:item[@"path"]]) {
+                    pathGood = true;
+                }
+            }
+        }
 
-            switch(connectionStatus){
-                case ALLOK:
-                    [self tryOpenProject:projectid];
-                    break;
-                case PERMISSION_DENIED:
-                    if (connectionAttempts == 0) {
-                        connectionAttempts = connectionAttempts + 1;
-                        NSDictionary *keychainData = [ProjectLockerAndKeychainFunctions load_data_from_keychain];
-                        
-                        [ProjectLockerAndKeychainFunctions login_to_project_server:[keychainData valueForKey:@"username"]
-                                                                          password:[keychainData valueForKey:@"password"]
-                                                                 completionHandler:^(enum ReturnValues loginResult) {
-                            if(loginResult!=ALLOK) {
-                                NSLog(@"Could not log in to projectlocker - error was %lu", (unsigned long)loginResult);
-                                if(loginResult==PERMISSION_DENIED) {
-                                    [self showError:@"Permission Denied" informativeText:@"Please check you have entered your username and password correctly." showPrefs:YES];
-                                }
-                            }
-                        } errorHandler:^(NSURLResponse *response, NSError *err) {
-                            [self setErrorAlert:[err localizedDescription]];
-                            NSLog(@"Could not log in to projectlocker: %@", [err localizedDescription]);
-                        }];
-                        break;
-                    }
-                    [self showError:@"Permission Denied" informativeText:@"Please check you have entered your username and password correctly." showPrefs:YES];
-                    break;
-                case SERVER_ERROR:
-                    [self showError:@"Server Error" informativeText:@"An error occurred on the server. Please report this error to multimediatech@theguardian.com" showPrefs:NO];
-                    break;
-                default:
-                    [self showError:@"Unknown Error" informativeText:@"An unknown error occurred. Please report this error to multimediatech@theguardian.com" showPrefs:NO];
-                    break;
+        NSArray *extensionsWhiteListArray = [[NSUserDefaults standardUserDefaults] arrayForKey:@"extensions_list"];
```

Comment:
See above, this should be replaced with another call to `checkIsInWhitelist`, using `@"paths_list"` for the `whiteListName`

Reply:
Done.